lib/request.ts

```diff
@@ -119,89 +130,114 @@ export const request = (
         originalReject(errorWithTimings);
       };
     }
-    const requestObject = httpRequestFn(
-      options,
-      (response: IncomingMessage) => {
+
+    const requestObject = httpRequestFn(options);
+
+    function setReadTimeout() {
+      requestObject.setTimeout(readTimeout, () => {
+        requestObject.socket.destroy();
+        reject(
+          new Error(
+            `Failed to read data within ${readTimeout} ms from ${options.host}`
```

Comment:
In practice (we run this work somewhere) this lead to logging undefined.. Seem like there's a bug %)

Reply:
I don't understand..can you give more context?

Reply:
We had these changes running in fsa and in practice this code `options.host` sometimes is `undefined` %)

I'm thinking maybe it's host vs hostname or smth

Reply:
It is `hostname`. I have verified it.

Reply:
if host is undefined, hostname would also be undefined AFAIR.. hostname = host without port.. We had this code running in tailor forever and i never seen undefined.. Did i miss something?

Reply:
oops, now i got it.. its perron logic of dealing with client options that makes this code undefined. 

You are right, host will be undefined when the url is not passed as string to ServiceClient. 

https://github.com/zalando-incubator/perron/blob/c06ab9fe8fb0c5cc7b675be67322db66de4718d3/lib/client.ts#L409-L410